contracts/token-distribution/SaleRounds.sol

```diff
@@ -204,115 +159,186 @@ contract SaleRounds is TokenDistribution, GameOwner, ERC20 {
         }
 
         reserveTokensInternal(roundType, _to, _amount);
+        emit ReserveTokensEvent(_roundType, _amount, _to);
     }
 
     // @_amount is going be decimals() == default(18) digits
-    function mintTokensForPublic(string calldata _roundType, address _to, uint _amount) public
-        onlyOwner isRoundActive(_roundType) {
+    function mintTokensForPublic(string calldata _roundType, address _to, uint _amount) external
+    onlyOwner {
         RoundType roundType = getRoundTypeByKey(_roundType);
 
         require(roundType == RoundType.PUBLIC , "round type is not valid");
-        require(roundDistribution[roundType].totalRemaining >= _amount, "total remaining amount is not enough" );
+        require(roundDistribution[roundType].totalRemaining >= _amount, "total remaining amount is not enough");
 
         roundDistribution[roundType].totalRemaining -= _amount;
         _mint(_to, _amount);
+        tokensClaimable = true;
```

Comment:
So this method is called for public, and then immediately all the claiming is allowed for everyone? 

This is a bug.

Please create a separate setter function that enables claiming (and also sets the time from where claiming is allowed)

Reply:
Or is this intentional that "vesting starts the moment the public tokens are being minted". In this case, please also note the vesting timestamp at that point. However, we need to understand if Jan wants the vesting timestamps to be fixed (e.g. pre-written into the individual allocations, with prederermined dates).

Reply:
Moved it to a separate setter method "setTokensToClaimable"